lib/ansible/plugins/httpapi/checkpoint.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2018 Cisco and/or its affiliates.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+httpapi : checkpoint
+short_description: HttpApi Plugin for Checkpoint devices
+description:
+  - This HttpApi plugin provides methods to connect to Checkpoint
+    devices over a HTTP(S)-based api.
+version_added: "2.8"
+"""
+
+import json
+import os
+import re
+
+from ansible.module_utils.basic import to_text
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible.plugins.httpapi import HttpApiBase
+from urllib3 import encode_multipart_formdata
+from urllib3.fields import RequestField
+from ansible.module_utils.connection import ConnectionError
+from ansible.utils.display import Display
+
+BASE_HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json'
+}
+
+display = Display()
+
+
+class HttpApi(HttpApiBase):
+    def __init__(self, connection):
+        super(HttpApi, self).__init__(connection)
+        self._auth = None
+
+    def login(self, username, password):
+        if username and password:
+            payload = {'user': username, 'password': password}
+            url = '/web_api/login'
+            response, response_data = self._send_auth_request(
+                url, json.dumps(payload), method='POST', headers=BASE_HEADERS
+            )
+            response = self._response_to_json(self._get_response_value(response_data))
+        else:
+            raise AnsibleConnectionFailure('Username and password are required for login')
+
+        try:
+            self._auth = response['sid']
```

Comment:
```suggestion
            self.connection._auth = {'X-chkp-sid': response['sid']}
```
Right, sorry, I told you the wrong thing for this